### ISSUE TITLE
Dynamic on-chain donation address

### ISSUE THREAD

**Issue by dadofsambonzuki** [teambtcmap/btcmap.org#109]
Use the [LNBits 'api_fresh_address_watchonly' API](https://core.btcmap.org/docs#/watchonly/api_fresh_address_watchonly_api_v1_address__wallet_id__get) to get dynamic on-chain addresses for donations.

**Comment by dadofsambonzuki** [comment]
We now have an issue with large gaps between addresses, which causes issues with wallets.

Need to re-think the strategy here, perhaps using a pool of addresses (maybe randomly served on page load), which are refreshed periodically.

**Comment by secondl1ght** [comment]
Maybe we can put in an issue with lnbits to add a new `getNextUnusedAddress` API endpoint. Seems like it should be an option there instead of always grabbing a new one.

**Comment by dadofsambonzuki** [comment]
Great idea. I can do that.

**Comment by dadofsambonzuki** [comment]
[Done.](https://github.com/lnbits/watchonly/issues/13).

In the meantime perhaps we could just have a button that generates an address when clicked rather than upon page load?

**Comment by dadofsambonzuki** [comment]
https://github.com/lnbits/watchonly/issues/13#issuecomment-2125087396

**Comment by dadofsambonzuki** [comment]
I think we need to progress with the button to generate an address ASAP @secondl1ght. Loading the wallet with these large gaps is almost impossible right now!

**Comment by secondl1ght** [comment]
Yeah we get around 650 views on that page per month so that is the rate the gap will increase. Keep in mind this won't solve the current gaps though and there still may be gaps in the future but not as many.

**Comment by secondl1ght** [comment]
I guess the other option is to go back to the static address.
